### DISCUSSION TITLE
AWS Neptune Sample: amazon-neptune-and-aws-cdk-for-amundsen

### DISCUSSION THREAD

**Discussion by ct-19** [community/community#63507] [Programming Help]
### Select Topic Area

Bug

### Feature Area

Issues

### Body

This CDK cannot deploy.  It fails when deploying the Amundsen-Blog-VPC-Stack when it is creating a lambda function tied to the S3BucketAutoDeleteObjects because it uses Nodejs.12  Line 1145 json file that is dropped in a S3 bucket on build.  Changing it to 18 don't work :)

https://github.com/aws-samples/amazon-neptune-samples/tree/master/amazon-neptune-and-aws-cdk-for-amundsen

  "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F": {
   "Type": "AWS::Lambda::Function",
   "Properties": {
    "Code": {
     "S3Bucket": {
      "Fn::Sub": "cdk-hnb659fds-assets-${AWS::AccountId}-${AWS::Region}"
     },
     "S3Key": "483ae06ed27ef8ca76e011264d772420593a6cfe8544759c306ef3b98c9e25be.zip"
    },
    "Timeout": 900,
    "MemorySize": 128,
    "Handler": "__entrypoint__.handler",
    "Role": {
     "Fn::GetAtt": [
      "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
      "Arn"
     ]
    },
    "Runtime": "nodejs12.x",
    "Description": {
     "Fn::Join": [
      "",
      [
       "Lambda function for auto-deleting objects in ",
       {
        "Ref": "AirflowS3Bucket101BB5A8"
       },
       " S3 bucket."
      ]
     ]
    }
   },
   "DependsOn": [
    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092"
   ],
   "Metadata": {
    "aws:cdk:path": "Amundsen-Blog-VPC-Stack/Custom::S3AutoDeleteObjectsCustomResourceProvider/Handler",
    "aws:asset:path": "asset.483ae06ed27ef8ca76e011264d772420593a6cfe8544759c306ef3b98c9e25be",
    "aws:asset:property": "Code"
   }

@don-simpson show Migration from v1 to v2 here:
https://github.com/aws-samples/amazon-neptune-samples/commit/660586270a8b0038dd2f24628c68b0c37116261a

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @ct-19 ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Projects and Issues category is solely related to conversations around the GitHub products Issues and Projects. 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by ct-19** [comment]
Appears a change was made and Node 16 is being used somewhere.  I know get this banner message on run:

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!!                                                                                                                  !!
!!  Node 16 is approaching end-of-life and will no longer be supported in new releases after 2023-09-11.            !!
!!  Please upgrade to a supported node version as soon as possible.                                                 !!
!!                                                                                                                  !!
!!  This software is currently running on node v16.20.1.                                                            !!
!!  As of the current release of this software, supported node releases are:                                        !!
!!  - ^20.0.0 (Planned end-of-life: 2026-04-30)                                                                     !!
!!  - ^18.0.0 (Planned end-of-life: 2025-04-30)                                                                     !!
!!  - ^16.3.0 (Planned end-of-life: 2023-09-11) [DEPRECATED]                                                        !!
!!                                                                                                                  !!
!!  This warning can be silenced by setting the JSII_SILENCE_WARNING_DEPRECATED_NODE_VERSION environment variable.  !!
!!                                                                                                                  !!
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

And the deploy now fails here

7:29:07 PM | CREATE_FAILED        | Custom::S3AutoDeleteObjects           | Airflow-S3-Bucket/...omResource/Default
Received response status [FAILED] from custom resource. Message returned: Error: Cannot find module 'aws-sdk'
Require stack:
- /var/task/index.js
- /var/task/__entrypoint__.js
- /var/runtime/index.mjs
at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
at Module._load (node:internal/modules/cjs/loader:922:27)
at Module.require (node:internal/modules/cjs/loader:1143:19)
at require (node:internal/modules/cjs/helpers:110:18)
at Object.<anonymous> (/var/task/index.js:1:108)
at Module._compile (node:internal/modules/cjs/loader:1256:14)
at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
at Module.load (node:internal/modules/cjs/loader:1119:32)
at Module._load (node:internal/modules/cjs/loader:960:12)
at Module.require (node:internal/modules/cjs/loader:1143:19) (RequestId: 371d15ea-707d-4008-95fe-60c58f25acaa)
7:29:09 PM | ROLLBACK_IN_PROGRESS | AWS::CloudFormation::Stack            | Amundsen-Blog-VPC-Stack
The following resource(s) failed to create: [S3VPCE826CF84B, AirflowS3BucketAutoDeleteObjectsCustomResource1D4C2BF5, EC2MessagesVPCEF730A8E4, SSMVPCE2392B759, SSMMessagesVPCE7B89305B, VPCPublicSubnet1NATGatew
ayE0556630, ECRVPCE33A7DFE0, ECRDockerVPCE3D601A7C]. Rollback requested by user.
7:29:09 PM | ROLLBACK_IN_PROGRESS | AWS::CloudFormation::Stack            | Amundsen-Blog-VPC-Stack
The following resource(s) failed to create: [S3VPCE826CF84B, AirflowS3BucketAutoDeleteObjectsCustomResource1D4C2BF5, EC2MessagesVPCEF730A8E4, SSMVPCE2392B759, SSMMessagesVPCE7B89305B, VPCPublicSubnet1NATGatew
ayE0556630, ECRVPCE33A7DFE0, ECRDockerVPCE3D601A7C]. Rollback requested by user.


Verified aws-sdk is installed.  Installed latest and also tried npm link aws-sdk
